agr_literature_service/api/crud/topic_entity_tag_crud.py

```diff
@@ -586,6 +587,46 @@ def show_all_reference_tags(db: Session, curie_or_reference_id, page: int = 1,
         return [get_tet_with_names(db, tag, curie_to_name) for tag in all_tet]
 
 
+def get_reference_tags(db: Session, mod_abbreviation: str, days_updated: int = 7):
+
+    current_date = datetime.now()
+    past_date = current_date - timedelta(days=int(days_updated))
+    last_date_updated = past_date.strftime("%Y-%m-%d")
+
+    rows = db.execute(f"SELECT cr.curie, tet.*, u.email "
+                      f"FROM cross_reference cr "
+                      f"JOIN topic_entity_tag tet ON cr.reference_id = tet.reference_id AND cr.curie_prefix = '{mod_abbreviation}' "
+                      f"JOIN topic_entity_tag_source tets ON tet.topic_entity_tag_source_id = tets.topic_entity_tag_source_id "
+                      f"JOIN users u ON tet.updated_by = u.id "
+                      f"WHERE tets.data_provider = '{mod_abbreviation}' "
```

Comment:
I'm not sure if we should filter by data_provider or secondary_data_provider here

Reply:
We have "secondary_data_provider_id" column in tet source table, but there is nothing in this column, but "data_provider" column contains mod abbreviations.

Reply:
That's the mod_id and we should use that column. I don't know why it's empty but it shouldn't be. data_provider is the original data provider, which can be something different than a mod_abbreviation if the data was originally generated by an external resource